megamek/src/megamek/client/ui/swing/lobby/LobbyActions.java

```diff
@@ -307,77 +302,39 @@ public void customizeMechs(Collection<Entity> entities) {
-            // send changes
-            for (Entity entity : entities) {
-                // If a LAM with mechanized BA was changed to non-mech mode, unload the BA.
-                if ((entity instanceof LandAirMech)
-                        && entity.getConversionMode() != LandAirMech.CONV_MODE_MECH) {
-                    for (Entity loadee : entity.getLoadedUnits()) {
-                        entity.unload(loadee);
-                        loadee.setTransportId(Entity.NONE);
-                        client().sendUpdateEntity(loadee);
-                    }
-                }
-
-                client().sendUpdateEntity(entity);
 
-                // Changing state to a transporting unit can update state of
-                // transported units, so update those as well
-                for (Transporter transport : entity.getTransports()) {
-                    for (Entity loaded : transport.getLoadedUnits()) {
-                        client().sendUpdateEntity(loaded);
-                    }
-                }
-
-                // Customizations to a Squadron can effect the fighters
-                if (entity instanceof FighterSquadron) {
-                    entity.getSubEntities().forEach(client::sendUpdateEntity);
-                }
-            }
+        if (editable && cmd.isOkay()) {
+            sendCustomizationUpdate(entities);
         }
+
         if (cmd.isOkay() && (cmd.getStatus() != CustomMechDialog.DONE)) {
             Entity nextEnt = cmd.getNextEntity(cmd.getStatus() == CustomMechDialog.NEXT);
             customizeMech(nextEnt);
         }
     }
 
     /**
-     *
-     * @param entity
+     * Shows the unit configuration dialog for the given unit.
+     * @param entity the unit to configure
      */
     public void customizeMech(Entity entity) {
-        if (!validateUpdate(Arrays.asList(entity))) {
+        if (!validateUpdate(List.of(entity))) {
             return;
         }
-        boolean editable = client().getBots().get(entity.getOwner().getName()) != null;
-        Client c;
-        if (editable) {
-            c = (Client) client().getBots().get(entity.getOwner().getName());
-        } else {
-            boolean localGM = localPlayer().isGameMaster();
-            editable |= localGM || entity.getOwnerId() == localPlayer().getId();
-            c = client();
-        }
-
-        // When we customize a single entity's C3 network setting,
-        // **ALL** members of the network may get changed.
-        Entity c3master = entity.getC3Master();
-        ArrayList<Entity> c3members = new ArrayList<>();
-        Iterator<Entity> playerUnits = c.getGame().getPlayerEntities(c.getLocalPlayer(), false).iterator();
-        while (playerUnits.hasNext()) {
-            Entity unit = playerUnits.next();
-            if (!entity.equals(unit) && entity.onSameC3NetworkAs(unit)) {
-                c3members.add(unit);
-            }
-        }
+
+        String ownerName = entity.getOwner().getName();

```

Comment:
should this block be an isEditable function?

Reply:
it's needs to get the client as well. But still true, I unified this also.